text/0000-grid-primitives.md

```diff
@@ -0,0 +1,605 @@
+would be constructed using the above protobuf messages.  The examples use a
+Python-like syntax.
+
+### Bytes
+
+A bytes value would be represented as follows:
+
+```
+PropertyDefinition(
+    name="user_data",
+    data_type=PropertyDefinition.DataType.Bytes,
+    description="Arbitrary serialized user data."
+)
+```
+
+Because this is a protobuf message, the default value for this field is an empty
+byte array.
+
+### Boolean
+
+A boolean value would be represented as follows:
+
+```
+PropertyDefinition(
+    name="is_enabled",
+    data_type=PropertyDefinition.DataType.BOOLEAN,
+    required=True,
+    description="Indicates that the containing struct is enabled."
+)
+```
+
+The value would be represented as:
+
+```
+PropertyValue(
+    name="is_enabled",
+    data_type=PropertyDefinition.DataType.BOOLEAN,
+    boolean_value=True
+)
+```
+
+Because this is a protobuf message, the default value for this field is `False`.
+
+### String
+
+A UTF-8 string value would be represented as follows:
+
+```
+PropertyDefinition(
+    name="title",
+    data_type=PropertyDefinition.DataType.STRING,
+    required=True,
+    description="A blog post title."
+)
+```
+
+The value would be represented as:
+
+```
+PropertyValue(
+    name="title"
+    data_type=PropertyDefinition.DataType.STRING,
+    string_value="My Very Nice Blog Example"
+)
+```
+
+Because this is a protobuf message, the default value for this field is the
+empty string.
+
+### Number
+
+An integer value would be represented as the following type:
+
+```
+PropertyDefinition(
+    name="quantity",
+    data_type=PropertyDefinition.DataType.NUMBER,
+    number_exponent=0,
+    required=True,
+    description="The count of values in this container"
+)
+```
+
+This example shows an instance of a quantity of 23:
+
+```
+PropertyValue(
+    name="quantity",
+    data_type=PropertyDefinition.DataType.NUMBER,
+    number_value=23,
+)
+```
+
+A fractional value would be represented as the following type:
+
+```
+PropertyDefinition(
+    name="price",
+    data_type=PropertyDefinition.DataType.NUMBER,
+    number_exponent=-2,
+    required=True,
+    description="The the price this object"
+)
+
+```
+
+This example shows an instance of a price with the value $0.23:
+
+```
+PropertyValue(
+    name="price",
+    data_type=PropertyDefinition.DataType.NUMBER,
+    number_value=23,
+)
+```
+
+Because this is a protobuf message, the default exponent is `0` when the schema
+is created. Likewise, the default value for this property instance is `0`.
+
+### Enum
+
+An enum value would be represented as:
+
+```
+PropertyDefinition(
+    name='color',
+    data_type=PropertyDefinition.DataType.ENUM,
+    enum_options=['white', 'red', 'green', 'blue', 'blacklight'],
+    required=True
+)
+``` 
+
+An instance of this enum would be as follows:
+
+```
+PropertyValue(
+    name='color',
+    data_type=PropertyDefinition.DataType.ENUM,
+    enum_value='white'
```

Comment:
This example is incorrect. `enum_value` takes an integer, not a string.

Reply:
Fixed.